sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/TryCast.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.spark.sql.catalyst.expressions.codegen._
+import org.apache.spark.sql.catalyst.expressions.codegen.Block._
+import org.apache.spark.sql.types.DataType
+
+/**
+ * A special version of [[AnsiCast]]. It performs the same operation (i.e. converts a value of
+ * one data type into another data type), but returns a NULL value instead of raising an error
+ * when the conversion can not be performed.
+ *
+ * When cast from/to timezone related types, we need timeZoneId, which will be resolved with
+ * session local timezone by an analyzer [[ResolveTimeZone]].
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(expr AS type) - Casts the value `expr` to the target data type `type`. " +
+    "This expression is identical to CAST with configuration `spark.sql.ansi.enabled` as " +
+    "true, except it returns NULL instead of raising an error. Note that the behavior of this " +
+    "expression doesn't depend on configuration `spark.sql.ansi.enabled`.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_('10' as int);
+       10
+      > SELECT _FUNC_(1234567890123L as int);
+       null
+  """,
+  since = "3.2.0",
+  group = "conversion_funcs")
```

Comment:
Since this expr is not registered in `FunctionRegistry`, this description is not automatically added in the SQL doc via the Python script. So, I think its better to document this new feature in the SQL syntax doc manually.

Reply:
Actually I plan to create docs for both CAST and TRY_CAST, even with ANSI CAST. Grouping them into one section.
I have created https://issues.apache.org/jira/browse/SPARK-34917 and https://issues.apache.org/jira/browse/SPARK-34918

Reply:
Nice! Thanks for letting me know.